src/main/java/fi/metatavu/soteapi/wordpress/events/EventListJob.java

```diff
@@ -0,0 +1,203 @@
+import java.util.Collections;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+
+import com.afrozaar.wordpress.wpapi.v2.Client;
+import com.afrozaar.wordpress.wpapi.v2.Wordpress;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import fi.metatavu.soteapi.settings.SystemSettingController;
+import fi.metatavu.soteapi.tasks.AbstractSoteApiTaskQueue;
+import fi.metatavu.soteapi.wordpress.WordpressConsts;
+import fi.metatavu.soteapi.wordpress.events.model.DateDetails;
+import fi.metatavu.soteapi.wordpress.events.model.Event;
+import fi.metatavu.soteapi.wordpress.events.model.Events;
+import fi.metatavu.soteapi.wordpress.events.model.Term;
+import fi.metatavu.soteapi.wordpress.tasks.AbstractListJob;
+
+public class EventListJob extends AbstractListJob<Event, EventListTask> {
+  
+  private static final String CONTEXT = "/tribe/events/v1";
+
+  @Inject
+  private Logger logger;
+  
+  @Inject
+  private EventListQueue eventListQueue;
+  
+  @Inject
+  private EventUpdateQueue eventUpdateQueue;
+
+  @Inject
+  private SystemSettingController systemSettingController;
+  
+  @Override
+  protected AbstractSoteApiTaskQueue<EventListTask> getQueue() {
+    return eventListQueue;
+  }
+  
+  @Override
+  protected String getEnabledSetting() {
+    return WordpressConsts.EVENTS_SYNC_ENABLED;
+  }
+  
+  @Override
+  protected EventListTask createTask(int page) {
+    EventListTask newTask = new EventListTask();
+    newTask.setPriority(Boolean.FALSE);
+    newTask.setUniqueId(String.format("wp-events-list-%d", page));
+    newTask.setPage(page);
+    return newTask;
+  }
+  
+  @Override
+  protected Wordpress getWordpressClient() {
+    String url = systemSettingController.getSettingValue(WordpressConsts.URL_SETTING);
+    if (StringUtils.isNotBlank(url)) {
+      return new Client(CONTEXT, url, "", "", WordpressConsts.USE_PERMALINK_ENDPOINT, WordpressConsts.DEBUG_CLIENT);
+    }
+    
+    return null;
+  }
+
+  /**
+   * Get total number of pages
+   * 
+   * @param postType type of wordpress post
+   * @return total number of pages for post type
+   */
+  @Override
+  protected int getNumberOfPages() {
+    Events events = listEvents(1);
+    if (events != null) {
+      return events.getTotalPages();
+    }
+    
+    return 0;
+  }
+
+  private Events listEvents(int page) {
+    try {
+      ResponseEntity<String> responseEntity = getWordpressClient().doCustomExchange(String.format("%s&%d", getEndPointUri(), page), HttpMethod.GET, String.class, new Object[0], null, null, null);
+      ObjectMapper objectMapper = new ObjectMapper();
+      return objectMapper.readValue(responseEntity.getBody(), Events.class);
+    } catch (IOException e) {
+      logger.error("Failed to list events", e);
+    };
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)